### PR TITLE
Added -/+ implementation in Time.Scala

### PR DESCRIPTION
**Pull request by naqvisat**
Added -/+ function implementation in Time.Scala.

### CODE REVIEW

File: shared/src/main/scala/pl/metastack/metatime/Time.scala [hunk 1]
```diff
@@ -18,13 +16,82 @@ trait Component extends Ordered[Component] {
...
+      timeComponents(4)
+    else if (dateComponents.contains(thisClass) || dateComponents.contains(otherClass))
+      dateComponents(1)
+    else if (timeComponents.contains(thisClass) && timeComponents.contains(otherClass))
+      timeComponents(4)
+    else
+      null
+  }
+  
+  def getTimeObject(comp : Component) : Time = {
```
Comment (tindzk) [inline shared/src/main/scala/pl/metastack/metatime/Time.scala]: Is there a particular reason why you don't use pattern matching here? How about:

``` scala
comp match {
  case hour: Hour => Time(hour = hour.h)
  ...
}
```

Also, we should not use `null`.

File: shared/src/main/scala/pl/metastack/metatime/Time.scala [hunk 2]
```diff
@@ -18,13 +16,82 @@ trait Component extends Ordered[Component] {
...
+            )
+        resultTime
+      case _ => null
+    }
+  }
+  
+  def -(other: Component): Component = {
+    operationResultType(other) match {
+      case "Time" => 
+        val thisTime : Time =  getTimeObject(this)
```
Comment (tindzk) [inline shared/src/main/scala/pl/metastack/metatime/Time.scala]: For local variables you should not specify a type.

File: shared/src/main/scala/pl/metastack/metatime/Time.scala [hunk 3]
```diff
@@ -18,13 +16,82 @@ trait Component extends Ordered[Component] {
...
+      case _ => null
+    }
+  }
 }
 
 trait Hour extends Period with Component {
   val h: Int
 
-  override def unix(): Unix = ???
+  def unix(): Unix = Unix( h * 60 * 60 * 1000 )
```
Comment (tindzk) [inline shared/src/main/scala/pl/metastack/metatime/Time.scala]: Keep the `override` modifier. Remove spaces before/after brackets.

**Comment by naqvisat** [comment]
Not needed any more because I just opened a new Pull request based on updated branch.

**Pull request closed without merging**